internal/characters/travelerdendro/lealotus.go

```diff
@@ -0,0 +1,284 @@
+package travelerdendro
+
+import (
+	"fmt"
+
+	"github.com/genshinsim/gcsim/pkg/core"
+	"github.com/genshinsim/gcsim/pkg/core/attributes"
+	"github.com/genshinsim/gcsim/pkg/core/combat"
+	"github.com/genshinsim/gcsim/pkg/core/event"
+	"github.com/genshinsim/gcsim/pkg/core/glog"
+	"github.com/genshinsim/gcsim/pkg/enemy"
+	"github.com/genshinsim/gcsim/pkg/gadget"
+	"github.com/genshinsim/gcsim/pkg/reactable"
+)
+
+type LeaLotus struct {
+	*gadget.Gadget
+	*reactable.Reactable
+	burstAtk        *combat.AttackEvent
+	char            *char
+	targetingRadius float64
+	hitboxRadius    float64
+	r               reactable.Reactable
+	//icd related
+	icdTagOnTimer       [enemy.MaxTeamSize][combat.ICDTagLength]bool
+	icdTagCounter       [enemy.MaxTeamSize][combat.ICDTagLength]int
+	icdDamageTagOnTimer [enemy.MaxTeamSize][combat.ICDTagLength]bool
+	icdDamageTagCounter [enemy.MaxTeamSize][combat.ICDTagLength]int
+}
+
+func (s *LeaLotus) transfigInit() {
+	s.Core.Events.Subscribe(event.OnQuicken, func(args ...interface{}) bool {
+		target := args[0].(combat.Target)
+		if target == s {
+			s.transfig(attributes.Electro)
+			return true
+		}
+		return false
+	}, "lealotus-electro")
+
+	s.Core.Events.Subscribe(event.OnBloom, func(args ...interface{}) bool {
+		target := args[0].(combat.Target)
+		if target == s {
+			s.transfig(attributes.Hydro)
+			return true
+		}
+		return false
+	}, "lealotus-hydro")
+
+	s.Core.Events.Subscribe(event.OnBurning, func(args ...interface{}) bool {
+		target := args[0].(combat.Target)
+		if target == s {
+			s.transfig(attributes.Pyro)
+			return true
+		}
+		return false
+	}, "lealotus-pyro")
+}
+
+func (s *LeaLotus) AuraContains(e ...attributes.Element) bool {
+	for ele := range e {
+		if s.Reactable.Durability[ele] <= reactable.ZeroDur {
+			return false
+		}
+	}
+	return true
+}
+
+func (c *char) newLeaLotusLamp(duration int) *LeaLotus {
+	s := &LeaLotus{}
+	x, y := c.Core.Combat.Player().Pos()
+	s.Gadget = gadget.New(c.Core, core.Coord{X: x, Y: y, R: 1}, combat.GadgetTypLeaLotus)
+	s.Duration = duration
+	// First hitmark is 37f after spawn, all other pre-transfig hits will be 90f between.
+	s.ThinkInterval = 37
+	s.Gadget.OnThinkInterval = s.OnThinkInterval
+
+	s.Reactable = &reactable.Reactable{}
+	s.Reactable.Init(s, c.Core)
+	s.Durability[reactable.ModifierDendro] = 20
+
+	s.char = c
+	s.char.burstTransfig = attributes.NoElement
+
+	s.targetingRadius = 8
+	s.hitboxRadius = 3
+
+	procAI := combat.AttackInfo{
+		ActorIndex: c.Index,
+		Abil:       "Lea Lotus Lamp (Q)",
```

Comment:
I'd remove the (Q) here and at the pyro explosion, it's clear what it is anyways